xarray/core/dataarray.py

```diff
@@ -1097,10 +1097,28 @@ def fillna(self, value):
         if utils.is_dict_like(value):
             raise TypeError('cannot provide fill value as a dictionary with '
                             'fillna on a DataArray')
-        out = self._fillna(value)
+        out = ops.fillna(self, value, join=join)
         out.attrs = self.attrs
         return out
 
+    def combine_first(self, other):
```

Comment:
The fact that `data_vars_join` could sensibly differ for `combine_first` (as you have implemented it) suggests that a second method is reasonable. Consistency with pandas is also a nice factor. So I guess I've come around!